scripts/bundlebuilder.py

```diff
@@ -41,6 +41,60 @@ def execute(cmd, raise_exception=True):
             return p.returncode, output
 
 
+class Fetcher(object):
+    """
+    Fetcher supports different protocols/means to reach the bundle source.
+    """
+
+    @staticmethod
+    def fetch(repo, branch=None):
+        """
+        Based on the protocol described at the beginning of repo calls
+        the right fetch_* method.
+        Args:
+            repo: repository where the bundle resides
+            branch: if the repo is a git repo, this is the branch
+
+        Returns: a directory where the bundle is
+
+        """
+        if repo.startswith('local:'):
+            path = repo[len('local:'):]
+            return Fetcher.fetch_local(path)
+        else:
+            return Fetcher.fetch_git(repo, branch)
+
+    @staticmethod
+    def fetch_local(path):
+        """
+        Just copy the bundle to a temp location
+        Args:
+            path: the path to the bundle
+
+        Returns: the temp directory containing the bundle
+
+        """
+        tempdir = mkdtemp()
+        destination = "{}/bundle".format(tempdir)
```

Comment:
The destination directory must not already exist or else the copytree will fail. So instead of dancing around the temp dirs I though It would be easier to just create a bundles path level. Either way everything inside the temp dir is deleted in the end of the bundle build/test cycle.